### ISSUE TITLE
allow to specify concatenation axis for buffered streamer

### ISSUE THREAD

**Issue by faroit** [pescadores/pescador#137]
pytorch defaults to `(seq_len, batch, input_size)` in their [recurrent modules](https://pytorch.org/docs/stable/nn.html#torch.nn.LSTM). Its possible to change this to batch first, but I measured a significant slowdown. 

Why not allow the user to provide a the axis for concatenation when using the buffered streamer for convenience.

something like

`pescador.buffer_stream(streamer, minibatch_size, axis=1)`

which could use `np.swapaxis` or some other tricks inside pescador to stack the samples, would do the trick.

**Comment by bmcfee** [comment]
Does this dupe #127?  (But yes, I agree, buffering should have an axis parameter)
